diff --git a/third_party/gpus/cuda/hermetic/cuda_nvrtc.BUILD.tpl b/third_party/gpus/cuda/hermetic/cuda_nvrtc.BUILD.tpl
index 7c0399a..17b1d78 100644
--- a/third_party/gpus/cuda/hermetic/cuda_nvrtc.BUILD.tpl
+++ b/third_party/gpus/cuda/hermetic/cuda_nvrtc.BUILD.tpl
@@ -10,16 +10,11 @@ cc_import(
     shared_library = "lib/libnvrtc.so.%{libnvrtc_version}",
 )
 
-cc_import(
-    name = "nvrtc_builtins",
-    shared_library = "lib/libnvrtc-builtins.so.%{libnvrtc-builtins_version}",
-)
 %{multiline_comment}
 cc_library(
     name = "nvrtc",
     %{comment}deps = [
         %{comment}":nvrtc_main",
-        %{comment}":nvrtc_builtins",
     %{comment}],
     %{comment}linkopts = cuda_rpath_flags("nvidia/cuda_nvrtc/lib"),
     visibility = ["//visibility:public"],
